### PR TITLE
Fix ADB/MTP detection & adds another needed partitions to backup-restore lists

### PR DESCRIPTION
**Pull request by lddnsk**
1. I Fix ADB/MTP detection in the Windows. In your version device detected as unknown. I used another config and vendor&product IDs. Now all functions worked properly with Motorola drivers.
2. I added more partitions to backup-restore lists such as vendor_boot, frp, oemowninfo, efs, modem, recovery. This is needed on some cases.
3. I use device id "Lenovo_TB320FC instead of "serialno" for backup folder ( /sdcard/TWRP/BACKUPS/serialno/ --> /sdcard/TWRP/BACKUPS/Lenovo_TB320FC/ ).

### CODE REVIEW

**Comment by polygraphene** [comment]
Thanks for sending the pull request. However, it looks like it needs a little cleanup before merge.
Your commits include many unrelated changes.

1. The device is properly detected on my Windows PC. Did you install the usb driver in android sdk? The driver should work.  
I defined mtp,adb mode as 0x18D1:0x4EE2. It is included in google's driver.  
`(android-sdk)\extras\google\usb_driver\android_winusb.inf`  
```
;Google Nexus (generic)
...
%CompositeAdbInterface%     = USB_Install, USB\VID_18D1&PID_4EE2&MI_01
```

See the following links for the installation instruction.
https://developer.android.com/studio/run/win-usb
https://developer.android.com/studio/run/oem-usb#InstallingDriver

Also, idVendor is already set by the following line. Not neccessary to reset on the init.recovery.usb.rc.
https://github.com/TeamWin/android_bootable_recovery/blob/a28a9f3867483fb5026601d954a3bf9059886b71/etc/init.rc#L130

For 2. and 3.:
Thanks. I didn't think much about paritions or serialno. I can merge it after clean up the commits.

**Review by polygraphene — CHANGES_REQUESTED** [review]
Unneccessary commit. Please exclude it.

**Review by polygraphene — COMMENTED** [review]
Only delete an empty line? Seems like unnecessary commit.

**Review by polygraphene — COMMENTED** [review]
It removes proper markdown paragraph. Unnecessary change.

**Review by polygraphene — COMMENTED** [review]
This change is for your repository. Not suitable for my repo.

**Comment by lddnsk** [comment]
I am so sorry for superfluous moment. Just I don't know how pull partial changes request.
This is my first attempt to work with git. Not all my actions is right method.

**Review by polygraphene — COMMENTED** [review]
Many unrelated changes. A necessary part should be just 2-3 lines.

**Review by polygraphene — COMMENTED** [review]
Not necessary for my repo.

**Comment by lddnsk** [comment]
> Google Nexus (generic)

Yes. device detected as Nexus. and ADB was dn't work. And MTP device detected as some COM port or similar...
I don't think is right idea use another device config. I use similar config from stock and it fully works. But if you think I mistaked - may be is it )

File: BoardConfig.mk [hunk 1]
```diff
@@ -140,14 +143,29 @@ TW_USE_FSCRYPT_POLICY := 1
...
+#	vendor.display.config@2.0
+#TW_SUPPORT_INPUT_AIDL_HAPTICS ⏩ Установите "true", чтобы включить создание makefile для определенного оборудования.
+#TW_SUPPORT_INPUT_AIDL_HAPTICS_FQNAME ⏩ Введите правильный путь к данному оборудованию, и это позволит скомпилировать необходимые файлы. В данном случае это файлы для тактильных вибраторов.
+#* В зависимости от типа оборудования опция может поддерживаться в режиме AIDL (hal format=" aidl ") или в режиме HIDL (hal format="hidl").
+#Пример
+#TW_SUPPORT_INPUT_1_2_HAPTICS := true
+
 TW_CUSTOM_CPU_TEMP_PATH := /sys/devices/virtual/thermal/thermal_zone41/temp
 
+TW_MTP_DEVICE := /dev/usb-ffs/mtp
```
Comment (lddnsk) [inline BoardConfig.mk:165]: This

**Comment by polygraphene** [comment]
> I am so sorry for superfluous moment. Just I don't know how pull partial changes request.

It would be easier to create a new branch and copy only necessary changes.
Create a branch on your repo for my latest change. And copy required parts from this PR.

> This is my first attempt to work with git. Not all my actions is right method.

It's okay. If you have any questions, please feel free to ask.

**Comment by polygraphene** [comment]
> > Google Nexus (generic)
> 
> Yes. device detected as Nexus. and ADB was dn't work. And MTP device detected as some COM port or similar... I don't think is right idea use another device config. I use similar config from stock and it fully works. But if you think I mistaked - may be is it )

Okay. My PC might have many unnecessary drivers. I will test on cleanly installed windows.

**Comment by polygraphene** [comment]
Worked without problem on cleanly installed Win11 24H2. I didn't install google's driver.

image
image
image

**Comment by lddnsk** [comment]
> Worked without problem on cleanly installed Win11 24H2. I didn't install google's driver.

I cannot check now on clean windows but I have installed stock lenovo/moto drivers. I think if device don't work WITH OFFICIAL drivers this is not normally.
Another computer worked with your version.

ADB no device and no portable device at all
adb
fastbootD no device
fastbootd
Only if I turn off MTP then ADB device detected

My config:
my
my_fastbootd

But another comp ADB no work in mode ADB+MTP .

I don't know how is better. I will more testing and at this moment I thinking my pull request slapdashed. I'l be back later, thank you for your answers  :)

**Comment by polygraphene** [comment]
Thank you for testing. The device seems to be detected by another driver than google or lenovo's one in your PC.

Check what driver is 

Double click `Google Nexus Adb Interface` then open `Event` tab.
Look file name of `*.inf` file then find the file in `C:\Windows\INF`.

Open `*.inf` in text editor, then the file might have information about what driver is detecting the device.

In cleanly installed windows, the driver `winusb` from Microsoft can detect the device. And adb+MTP is working flawlessly.
image
image

After installing Lenovo Software Fix, it is detected as `LeMobile Android Device`. And adb+MTP is still working.
image
image
image

**Comment by polygraphene** [comment]
I thought my config should work on any driver, but if your config (Lenovo's idProduct?) is working fine for broader environment than mine, I would prefer yours.
I want to clearify the difference of your config and mine. And I want to know the real cause of the issue before merging PR.

**Pull request closed without merging**